src/Build/Evaluation/Expander.cs

```diff
@@ -2022,9 +2045,17 @@ ExpanderOptions options
                     return true;
                 }
 
+                int startLength = builder.Length;
+                bool truncate = (options & ExpanderOptions.Truncate) != 0;
+
                 // if the capture.Separator is not null, then ExpandExpressionCapture would have joined the items using that separator itself
                 foreach (var item in itemsFromCapture)
                 {
+                    if (truncate && (builder.Length - startLength) >= DefaultTruncationLimit)
```

Comment:
If the expansion is over 1024 chars, this approach will still keep the first 1024. I'm wondering whether it would be better to add an additional truncation strategy: keep only the first 3 items when truncate is true, like #5315 suggests.

Reply:
What if I changed this PR to only add `ExpanderOptions.TruncateItems`? And it could just skip items over 3, we wouldn't need to check string length against 1024 at all.

How often have we seen properties/metadata over 1024 chars?

Reply:
No idea on metadata, but solution files inject humongous global properties with xml in them (I thought there was an issue on that but can't find it anymore). I would keep your current approach and also add keeping only the first 3 items to further cut down on item size. `ExpanderOptions.TruncateItems` doesn't seem like it would add much value beyond just `ExpanderOptions.Truncate`. We can add it later if the need arises.

Reply:
I added a `DefaultTruncationItemLimit` and reworked things a bit.